src/universal/components/SignInPage/SignInPage.js

```diff
@@ -148,56 +112,60 @@ class SignInPage extends Component<Props, State> {
 
   renderLoadingError = () => {
     return (
-      <PageContainer>
-        <Header />
-        <ErrorContainer>
-          <h1>
-            🤭 Oops!
-          </h1>
-          <p>
-            We had some trouble signing you in!
-          </p>
-          <p>
-            Try going back to the <Link to="/signin" onClick={this.resetState}>Sign in page</Link> in order to sign in again.
-          </p>
-        </ErrorContainer>
-      </PageContainer>
+      <Fragment>
+        <h1>
+          🤭 Oops!
+        </h1>
+        <p>
+          We had some trouble signing you in!
+        </p>
+        <p>
+          Try going back to the <Link to="/signin" onClick={this.resetState}>Sign in page</Link> in order to sign in again.
+        </p>
+      </Fragment>
+    );
+  };
+
+  renderSignIn = () => {
+    const {error, submittingCredentials} = this.state;
+    return (
+      <SignIn
+        authProviders={THIRD_PARTY_AUTH_PROVIDERS}
+        getHandlerForThirdPartyAuth={this.getHandlerForThirdPartyAuth}
+        handleSubmitCredentials={this.handleSubmitCredentials}
+        error={error && error.error_description}
+        isSubmitting={submittingCredentials}
+      />
     );
   };
 
   render() {
     const {loggingIn, error} = this.state;
 
+    let pageContent: Node;
     if (loggingIn && !error) {
-      return this.renderLoading();
-    }
-    if (loggingIn && error) {
-      return this.renderLoadingError();
+      pageContent = this.renderLoading();
+    } else if (loggingIn && error) {
+      pageContent = this.renderLoadingError();
+    } else {
+      pageContent = this.renderSignIn();
     }
     return (
-      <PageContainer>
-        <Header />
-        <SignIn
-          hasError={Boolean(error)}
-          authProviders={this.authProviders}
-          getHandlerForThirdPartyAuth={this.getHandlerForThirdPartyAuth}
-          handleSubmitCredentials={this.handleSubmitCredentials}
-        />
-      </PageContainer>
+      <AuthPage title="Sign In | Parabol">
+        {pageContent}
+      </AuthPage>
     );
   }
 }
 
-const mapStateToProps = (state) => ({
-  hasSession: Boolean(state.auth.obj.sub)
-});
-
 const mapDispatchToProps = (dispatch) => ({
```

Comment:
remove. `connect()` gives you dispatch. this is for wrapping your actions in a dispatch (which is a horrid pattern, anyways).

Reply:
Haha, I actually think it's a fantastic pattern, as it allows you to keep react components unaware of the redux api.